### PR TITLE
move models to own files/simplify view

### PR DESCRIPTION
**Pull request by alessandrofelder**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

It is a step on the way towards being able to download new atlases, and update local atlases to newer versions, through a napari GUI.

**What does this PR do?**
This PR does two things (I should have split this into two PRs - sorry!) in preparation for moving atlas download functionality to a different widget/adding atlas update functionality in the GUI.

### Make the model/views more flexible to allow several (kinds of) views on the same model
* separates the datamodels for atlases and atlas regions ("structures") into separate files (they were in the view file before).
* Renames `AtlasTableView` to `AtlasViewerView`, which is the view that allows users to "view" atlases in `napari`. This paves the way to have other atlas views, e.g. an `AtlasUpdaterView` or so, which is a view that allows updating the atlas version, using the same data model.
* Removes functionality to download atlases in the current `BrainrenderWidget`, as this will be for visualisation purposes only.

### Minor refactor of the atlas table model

This PR does a minor refactor of `AtlasTableModel` (see below for details). 
* allows the model to be updated via the atlas api(e.g. when new atlas has been downloaded) without needing re-instantiation
* beautifies how the atlas name is displayed.

I've sign-posted these new functions [with a comment](https://github.com/brainglobe/brainrender-napari/pull/123/files#r1431124406).
The data model code has not changed otherwise (beyond moving files).

## References

Will facilitate #21 by separating concerns further.

## How has this PR been tested?

Existing tests refactored in line with changes detailed above, pass locally and on CI.

## Is this a breaking change?

Yes, in the sense that downloading atlases will now not be possible via the viewer widget.

## Does this PR require an update to the documentation?

Once #21 is addressed, yes, we'll need to update https://brainglobe.info/tutorials/visualise-atlas-napari.html

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [n/a] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: brainrender_napari/data_models/atlas_table_model.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    [name, self._format_name(name), "n/a", latest_version]
+                )
+
+        self._data = data
+
+    def _format_name(self, name: str) -> str:
+        formatted_name = name.split("_")
+        formatted_name[0] = formatted_name[0].capitalize()
+        formatted_name[-1] = f"({formatted_name[-1].split('um')[0]} \u03BCm)"
+        return " ".join([formatted for formatted in formatted_name])
```
Comment (alessandrofelder) [inline brainrender_napari/data_models/atlas_table_model.py:46]: These are the new functions added to the atlas table model.

**Review by IgorTatarnikov — CHANGES_REQUESTED** [review]
Looks good, just a few comments on my end!

File: brainrender_napari/widgets/atlas_viewer_view.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        self.customContextMenuRequested.connect(
+            self._on_context_menu_requested
+        )
+
+        self.doubleClicked.connect(self._on_row_double_clicked)
+        self.selectionModel().currentChanged.connect(self._on_current_changed)
+
+        for hidden_column in [0, 2, 3]:
+            # hide raw name(0), local and latest versions(2,3) in this view
+            self.hideColumn(hidden_column)
```
[the remark this answers is not in the document]
Comment (alessandrofelder) [reply brainrender_napari/widgets/atlas_viewer_view.py:55]: Good point - I can implement this with a simple refactor of our custom model.

File: brainrender_napari/data_models/atlas_table_model.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                        name,
+                        self._format_name(name),
+                        get_local_atlas_version(name),
+                        latest_version,
+                    ]
+                )
+            else:
+                data.append(
+                    [name, self._format_name(name), "n/a", latest_version]
+                )
```
[the remark this answers is not in the document]
Comment (alessandrofelder) [reply brainrender_napari/data_models/atlas_table_model.py:38]: Don't think it currently returns something sensible unfortunately - I've tested locally and it gives a list index out of range error. I've opened https://github.com/brainglobe/bg-atlasapi/issues/187 as a possibility for the future.

**Review by IgorTatarnikov — APPROVED** [review]
Looks good 👍

**Pull request merged**